buildscripts/platform_detection.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: GPL-3.0-or-later
+#     Copyright (c) 2022 by The qTox Project Contributors
+
+
+set -euo pipefail
+
+usage()
+{
+    echo "Download and build $DEP_NAME for the Windows cross compiling environment"
```

Comment:
maybe nitpick but the filename doesn't seem to match the usage message

Reply:
I see it's something the dep downloader sources. Seems fine then.

Reply:
after offline discussion I think this file may be due for a rename - since it provides more than one function, but it is only named after one function. will approve once this is addressed.

Reply:
I've updated the DEP_NAME to be passed directly to parse_arch, so that it's not passing anything through global state. File also renamed from `platform_detection.sh` to `build_utils.sh`.